support/reftest/reftest.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+"""
+
+Prepare reference test data for Exodus-lambda integration test.
+
+"""
+
+import os
+import sys
+import yaml
+import boto3
+
+import hashlib
+import requests
+import argparse
+
+from tqdm import tqdm
+from datetime import datetime, timezone
+
+
+class DBHandler:
+    def __init__(
+        self,
+        table,
+        access_id=None,
+        access_key=None,
+        session_token=None,
+        region_name=None,
+    ):
+        self.dynamodb = boto3.Session(
+            aws_access_key_id=access_id,
+            aws_secret_access_key=access_key,
+            aws_session_token=session_token,
+            region_name=region_name,
+        ).client("dynamodb")
+
+        self.table = table
+
+    def search_item_from_web_uri(self, web_uri):
+        return self.dynamodb.query(
+            TableName=self.table,
+            Limit=1,
+            ScanIndexForward=False,
+            KeyConditionExpression="web_uri = :u and from_date <= :d",
```

Comment:
Should we really have the from_date <= check here?  This is used during "clean", right?

To me, it doesn't really make sense right now.  You made it possible for "prepare" to upload the data with a date in the future, but made it impossible for "clean" to clean that data.  Shouldn't they be consistent?  And why should the semantics of "clean" be only to delete the most recent non-future entry for any given URL?  (Which is not really "clean", it's "roll back to previous version").

I'm not quite understanding what is the intended behavior of "clean" and when I'd want to use that.